### ISSUE TITLE
In pull #2, ClassCastException is thrown very late when operator gets types wrong

### ISSUE THREAD

**Issue by nachivpn** [nachivpn/molr-core#3]
In pull #2 The issue is detected when the mission is:
* about to be executed if input type is wrong
* after execution (!) when the output type is wrong

Can this be detected earlier in both cases?

Example:

```java
CompletableFuture<RunMissionController<String>> futureController = mExecService.<Integer, String>runToCompletion("cern.molr.sample.IntDoubler", 21);
        try {
            RunMissionController<String> controller = futureController.get();
            CompletableFuture<String> futureResult = controller.getResult();
            return futureResult.get();
        } catch (Exception e) {
            System.err.println("Operator received an error:");
            e.printStackTrace();
            throw e;
        }
```
gives:
```java
Operator received an error:
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
	at cern.molr.sample.SampleOperator.operatorRun2(SampleOperator.java:47)
	at cern.molr.sample.SampleMain.main(SampleMain.java:18)
Exception in thread "main" java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
	at cern.molr.sample.SampleOperator.operatorRun2(SampleOperator.java:47)
	at cern.molr.sample.SampleMain.main(SampleMain.java:18)
```
and 
```java
 CompletableFuture<RunMissionController<Integer>> futureController = mExecService.<String, Integer>runToCompletion("cern.molr.sample.IntDoubler", "21");
        try {
            RunMissionController<Integer> controller = futureController.get();
            CompletableFuture<Integer> futureResult = controller.getResult();
            return futureResult.get();
        } catch (Exception e) {
            System.err.println("Operator received an error:");
            e.printStackTrace();
            throw e;
        }
```
gives:
```java
Operator received an error:
java.util.concurrent.ExecutionException: java.util.concurrent.CompletionException: BAD argument! Wrong type?
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at cern.molr.sample.SampleOperator.operatorRun2(SampleOperator.java:47)
	at cern.molr.sample.SampleMain.main(SampleMain.java:18)
Caused by: java.util.concurrent.CompletionException: BAD argument! Wrong type?
	at cern.molr.sample.SampleSupervisor$RunMoleSpawner.spawnMoleRunner(SampleSupervisor.java:117)
	at cern.molr.sample.SampleSupervisor$SampleMoleSupervisor.lambda$0(SampleSupervisor.java:45)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1582)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer
	at cern.molr.sample.IntegerFunctionMole.run(IntegerFunctionMole.java:1)
	at cern.molr.sample.SampleSupervisor$RunMoleSpawner.spawnMoleRunner(SampleSupervisor.java:115)
	... 7 more
Exception in thread "main" java.util.concurrent.ExecutionException: java.util.concurrent.CompletionException: BAD argument! Wrong type?
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at cern.molr.sample.SampleOperator.operatorRun2(SampleOperator.java:47)
	at cern.molr.sample.SampleMain.main(SampleMain.java:18)
Caused by: java.util.concurrent.CompletionException: BAD argument! Wrong type?
	at cern.molr.sample.SampleSupervisor$RunMoleSpawner.spawnMoleRunner(SampleSupervisor.java:117)
	at cern.molr.sample.SampleSupervisor$SampleMoleSupervisor.lambda$0(SampleSupervisor.java:45)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1582)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Integer
	at cern.molr.sample.IntegerFunctionMole.run(IntegerFunctionMole.java:1)
	at cern.molr.sample.SampleSupervisor$RunMoleSpawner.spawnMoleRunner(SampleSupervisor.java:115)
	... 7 more

```

**Comment by nachivpn** [comment]
`TypeToken` from Guava could help?

**Comment by nachivpn** [comment]
Run-time type safety cannot be guaranteed using generics alone. The question is how early can this run-time error be detected..

**Comment by nachivpn** [comment]
Addressed in #4
